include/net/zzhc.rst

```diff
@@ -0,0 +1,74 @@
+.. _lib_zzhc:
+
+China Telecom ZZHC library
+#################################################
+
+The China Telecom ZZHC library implements the self-registration (Zi-ZHu-Ce, ZZHC) functionality that registers the module information on the server when the network is available.
+
+Overview
+********
+
+The telecommunication modules that are part of the China Telecom network are required to provide certain information stored on the module to the `zzhc.vnet.cn` server.
+This is required for conformity with Chapter 6 of the China Telecom IoT Module Requirements Whitepaper.
+
+The ZZHC library is started as a background thread on boot and it automatically detects the availability of the operator network.
+Once the Internet connectivity is ready and a new SIM card is detected on boot, the library proactively uploads the following information to the ZZHC server:
+- IMEI
+- model number (that is, nRF9160)
+- modem revision
+- nRF9160 SIM ICCID
+- nRF9160 SIM IMSI
+On any failure, which include unknown/negative response from server or any error during program execution, the ZZHC library retries the next hour for at most 5 times.
+
+.. _lib_zzhc_dependencies:
+
+Dependencies
+************
+
+The ZZHC library only when CONFIG_TRUSTED_EXECUTION_NONSECURE is enabled (i.e. Trusted Execution: Non-Secure firmware image).
+
+The ZZHC library requires the following to operate, therefore they are automatically selected when ZZHC library is enabled.
+   * CONFIG_BSD_LIBRARY=y
+   * CONFIG_SETTINGS=y
+   * CONFIG_AT_CMD=y
+   * CONFIG_AT_CMD_PARSER=y
+   * CONFIG_AT_NOTIF=y
+   * CONFIG_BASE64=y
+   * CONFIG_JSON_LIBRARY=y
+
+.. _lib_zzhc_configuration:
+
+Configurations
+**************
+
+The ZZHC library is by default not used in any sample.
+To enable the library so that the end product can be approved by China Telecom, enable Kconfig option CONFIG_ZZHC.
+In prj.conf, set CONFIG_ZZHC=y.
+
+The thread behavior can also be configured using Kconfig options:
```

Comment:
Do I understand correctly these two thread behavior options are not mandatory?

Reply:
No, priority and stack size are supposed to be tweaked by SDK users.

Reply:
OK, no further edits needed.